userspace/falco/app/actions/configure_interesting_sets.cpp

```diff
@@ -0,0 +1,195 @@
+	{
+		return;
+	}
+
+	/* Get the names of the events (syscall and non syscall events) that were not activated and print them. */
+	auto names = libsinsp::events::sc_set_to_names(unsupported_sc_set);
+	std::cerr << "Loaded rules match event types that are not activated or unsupported with current configuration: warning (unsupported-evttype): " + concat_set_in_order(names) << std::endl;
+	std::cerr << "If syscalls in rules include high volume I/O syscalls (-> activate via `-A` flag), else syscalls might be associated with syscalls undefined on your architecture (https://marcin.juszkiewicz.com.pl/download/tables/syscalls.html)" << std::endl;
+}
+
+static void select_event_set(falco::app::state& s, const libsinsp::events::set<ppm_sc_code>& rules_sc_set)
+{
+	/* PPM syscall codes (sc) can be viewed as condensed libsinsp lookup table
+	 * to map a system call name to it's actual system syscall id (as defined
+	 * by the Linux kernel). Hence here we don't need syscall enter and exit distinction. */
+	auto rules_names = libsinsp::events::sc_set_to_names(rules_sc_set);
+	if (!rules_sc_set.empty())
+	{
+		falco_logger::log(LOG_DEBUG, "(" + std::to_string(rules_names.size())
+			+ ") events in rules: " + concat_set_in_order(rules_names) + "\n");
+	}
+
+	/* DEFAULT OPTION:
+	* Current sinsp_state_sc_set() approach includes multiple steps:
+	* (1) Enforce all positive syscalls from each Falco rule
+	* (2) Enforce static `libsinsp` state set (non-adaptive, not conditioned by rules,
+	* but based on PPME event table flags indicating generic sinsp state modifications)
+	* -> Final set is union of (1) and (2)
+	*
+	* Fall-back if no valid positive syscalls in "base_syscalls",
+	* e.g. when using "base_syscalls" only for negative syscalls.
+	*/
+	auto base_sc_set = libsinsp::events::sinsp_state_sc_set();
+
+	/* USER OVERRIDE INPUT OPTION "base_syscalls". */
+	std::unordered_set<std::string> user_positive_names = {};
+	std::unordered_set<std::string> user_negative_names = {};
+	extract_base_syscalls_names(s.config->m_base_syscalls, user_positive_names, user_negative_names);
+	auto user_positive_sc_set = libsinsp::events::names_to_sc_set(user_positive_names);
+	auto user_negative_sc_set = libsinsp::events::names_to_sc_set(user_negative_names);
+
+	if (!user_positive_sc_set.empty())
+	{
+		// user overrides base event set
+		base_sc_set = user_positive_sc_set;
+
+		// we re-transform from events set to names to make
+		// sure that bad user inputs are ignored
+		auto user_positive_sc_set_names = libsinsp::events::sc_set_to_names(user_positive_sc_set);
+		falco_logger::log(LOG_DEBUG, "+(" + std::to_string(user_positive_sc_set_names.size())
+			+ ") syscalls added (base_syscalls override): "
+			+ concat_set_in_order(user_positive_sc_set_names) + "\n");
+	}
+
+	// selected events are the union of the rules events set and the
+	// base events set (either the default or the user-defined one)
+	s.selected_sc_set = rules_sc_set.merge(base_sc_set);
```

Comment:
I like how now that we reverted to starting from sc_set we could simplify and clean up "base_syscalls" right away and be more consistent 🎉